CESQLParser.g4

```diff
@@ -0,0 +1,103 @@
+grammar CESQLParser;
+
+import CESQLLexer;
+
+// Entrypoint
+cesql: expression EOF;
+
+// Structure of operations, function invocations and expression
+functionInvocation
+    : functionIdentifier LR_BRACKET functionParameterList RR_BRACKET
+    ;
+
+unaryOperation
+    : unaryLogicOperator expression #unaryLogicExpression
+    | unaryNumericOperator expression # unaryNumericExpression
+    ;
+
+expression
+    : functionInvocation #functionInvocationExpression
+    | unaryOperation #unaryExpression
+    // LIKE, EXISTS and IN takes precedence over all the other operators
+    | expression notOperator? likeOperator stringLiteral #likeExpression
+    | existsOperator identifier #existsExpression
```

Comment:
what about inlining the `xxOperator` rules? example: `expression NOT? LIKE stringLiteral`. I find this easier to read and it's consistent with what you did below.